jdaviz/configs/default/plugins/model_fitting/model_fitting.py

```diff
@@ -470,6 +465,17 @@ def _initialize_model_component(self, model_comp, comp_label, poly_order=None):
                      "parameters": [], "model_kwargs": {}}
         model_cls = MODELS[model_comp]
 
+        # Need to set the units the first time we initialize a model component, after this
+        # we listen for display unit changes
+        if (self._units is None or self._units == {} or 'x' not in self._units or
+                'y' not in self._units):
+            uc = self.app._jdaviz_helper.plugins['Unit Conversion']
+            self._units['x'] = uc.spectral_unit.selected
+            if self.cube_fit:
+                self._units['y'] = uc._obj.sb_unit_selected
+            else:
+                self._units['y'] = uc.flux_unit.selected
```

Comment:
eventually this probably will be safe, but might need to use `app._get_display_units` for now to get the fallbacks when unit conversion does not exist

Reply:
for non-cube fit, this also needs to listen to `spectral_y_type` (`app._get_display_units('spectral_y')` handles that logic as well).